Don't muck with the stdlib. MacPorts already sets it the way we want.
https://github.com/aseprite/aseprite/commit/be7c26da5eb29dd50705babc4277cb8ebce9af91
--- aseprite/CMakeLists.txt.orig	2022-01-12 08:10:33.000000000 -0600
+++ aseprite/CMakeLists.txt	2022-01-30 22:43:28.000000000 -0600
@@ -30,12 +30,6 @@
 # Aseprite project
 project(aseprite C CXX)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
-  # As we compile with CMAKE_OSX_DEPLOYMENT_TARGET=10.9, we have to
-  # explicitly say that we want to use libc++ instead of the GNU
-  # libstdc++
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
-endif()
 
 # Check repository status
 if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/laf/CMakeLists.txt)
